bin/create-map-file

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import os
+import click
+import click_log
+import logging
+
+import ROOT
+from rootpy.io import root_open
+from rootpy.tree import Tree
+from cmsl1t.utils.module import load_L1TNTupleLibrary
+import cmsl1t
+
+import yaml
+import collections
+
+logger = logging.getLogger(__name__)
+logging.getLogger("rootpy.tree.chain").setLevel(logging.WARNING)
+click_log.basic_config(logger)
+
+ALIAS_REGISTRY = []
+
+
+def extract_branches(tree):
+    paths = []
+    for branch in tree.branches:
+        branchName = branch.GetName()
+
+        if isinstance(branch, ROOT.TBranchElement):
+            leaves = ['.'.join([branchName, l.GetName()])
+                      for l in branch.GetListOfBranches()]
+            if leaves:
+                paths += leaves
+            else:
+                paths.append(branchName)
+
+    return sorted(paths, key=lambda s: s.lower())
+
+
+def extract_trees(input_file):
+    trees = {}
+
+    with root_open(input_file) as f:
+        for path, dirs, objects in f.walk():
+            for objName in objects:
+                objPath = os.path.join(path, objName)
+                obj = f.Get(objPath)
+                if isinstance(obj, Tree):
+                    trees[objPath] = dict(
+                        name=objName,
+                        branches=extract_branches(obj),
+                    )
+        return trees
+
+
+def convert_to_dict(trees):
+    trees_dict = {}
+    for path, tree in trees.items():
+        trees_dict[path] = dict(
+            name=tree['name'],
```

Comment:
I think the tree name could be a bit misleading, since it can be the same for different trees. `L1CaloTowerTree` is one example -- it appears twice in the current yaml file.  The path might actually be more helpful, since it's more unique?

Reply:
Indeed, once as
```yaml
l1CaloTowerEmuTree/L1CaloTowerTree:
    name: L1CaloTowerTree
    optional: false
    branches:
      CaloTP.ecalTPCaliphi:
        aliases:
        - event.l1CaloTowerEmuTree_L1CaloTowerTree_CaloTP_ecalTPCaliphi
        - event.emu_L1CaloTowerTree_CaloTP_ecalTPCaliphi
        - event.emu_CaloTP_ecalTPCaliphi
```
and once without the `emu` bits. As you see the path is included at the top level description, the `name` entry is obsolete in the current implementation. My thought was that it might be needed if you want to refer to the treename itself without the need for `path.split('/')[-1]`